jina/types/request/data.py

```diff
@@ -157,32 +168,33 @@ def proto_wo_data(
         calling :meth:`SerializeToString`.
         :return: protobuf instance containing parameters
         """
-        if self.is_decompressed:
-            return self._pb_body
-        elif self.is_decompressed_wo_data:
-            return self._pb_body_wo_data
-        else:
+        if self._pb_body is None:
             self._decompress_wo_data()
-            return self._pb_body_wo_data
+        return self._pb_body
 
     @property
-    def proto(self) -> 'jina_pb2.DataRequestProto':
+    def proto(
+        self,
+    ) -> Union['jina_pb2.DataRequestProto', 'jina_pb2.DataRequestProtoWoData']:
         """
         Cast ``self`` to a :class:`jina_pb2.DataRequestProto`. Laziness will be broken and serialization will be recomputed when calling.
-        Under the hood it is calling :meth:`jina.types.request.data.DataRequest.proto_wo_data`. This method is keep for legacy.
         :meth:`SerializeToString`.
         :return: DataRequestProto protobuf instance
         """
-        return self.proto_data
+        if not self.is_decompressed:
+            self._decompress()
+        return self._pb_body
```

Comment:
I think this could return either a proto with or without data, but the docstring doesn't match that behaviour (the type hint does, though).